src/jit/codegenarmarch.cpp

```diff
@@ -1381,7 +1386,17 @@ void CodeGen::genCodeForCpBlk(GenTreeBlk* cpBlkNode)
     }
 #endif // _TARGET_ARM64_
 
+    if (cpBlkNode->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
A comment stating that we place a full memory barrier before and after the cpblk operation, should be added here